From 440c7d39e1a9e13c7f4bf38e88cf3a2dcb4720b8 Mon Sep 17 00:00:00 2001
From: Chun-wei Fan <fanchunwei@src.gnome.org>
Date: Fri, 5 Jun 2015 17:37:41 +0800
Subject: [PATCH 5/5] src/gen_dispatch.py: Make Generated Code Build Under C89

...so that the generated code are buildable by pre-2013 Visual Studio.  The
main thing that this does is that we avoid named initializers, but instead
initialize the structs in old-school C89 way.

The generated code may not look that robust, but since this is generated
code, I think this is not an issue; when the Khronos registry gets updated,
all that is needed is that the code gets re-generated, and we have the
items in the right order.

These are the only changes required, after getting the code to work with
Visual Studio 2013, for libepoxy to build with Visual Studio 2008-2012,
provided that the user finds a suitable implementation of stdint.h,
inttypes.h and stdbool.h.
---
 src/gen_dispatch.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/gen_dispatch.py b/src/gen_dispatch.py
index 1cad4ff..bd9e424 100755
--- a/src/gen_dispatch.py
+++ b/src/gen_dispatch.py
@@ -643,12 +643,13 @@ class Generator(object):
         # for epoxy_print_failure_reasons().
 
         self.outln('static const char *enum_strings[] = {')
+        self.outln('    "", /* {0}_provider_terminator */'.format(self.target))
 
         sorted_providers = sorted(self.provider_enum.keys())
 
         for human_name in sorted_providers:
             enum = self.provider_enum[human_name]
-            self.outln('    [{0}] = "{1}",'.format(enum, human_name))
+            self.outln('    "{1}", /* {0} */'.format(enum, human_name))
         self.outln('};')
         self.outln('')
 
@@ -768,7 +769,7 @@ class Generator(object):
 
         self.outln('static struct dispatch_table resolver_table = {')
         for func in self.sorted_functions:
-            self.outln('    .{0} = epoxy_{0}_dispatch_table_rewrite_ptr,'.format(func.wrapped_name))
+            self.outln('    epoxy_{0}_dispatch_table_rewrite_ptr, /* {0} */'.format(func.wrapped_name))
         self.outln('};')
         self.outln('')
 
-- 
2.3.5.windows.8

